Automattic/woocommerce-payments
# Added logic to format the example rate differently if a zero decimal currency is used

Fixes #2316 

#### Changes proposed in this Pull Request









- When a zero-decimal currency is used as the default store currency, show the exchange rates on the main enabled currencies differently (see example photos below). This is because the display of currency rates is currently formatted to 2DP, meaning when converting from a zero-decimal to non-zero-decimal rate, the rate would usually be displayed as 1.00 -> 0.00 (because of the high precision of the conversion rate).
- Display the rate in the same way in the add currrencies form, if the store uses a zero-decimal currency as default.

*Old display for Zero Decimal Currencies*:

<img width="737" alt="Screenshot 2021-07-05 at 15 33 22" src="https://user-images.githubusercontent.com/11770181/124487231-5c6eb180-dda6-11eb-8966-91006e54b1b2.png">

*New display for Zero Decimal Currencies*:

<img width="732" alt="Screenshot 2021-07-05 at 15 31 12" src="https://user-images.githubusercontent.com/11770181/124487224-5a0c5780-dda6-11eb-9fc5-a8871cb3c4a5.png">

#### Testing instructions





* Point your local instance of the client to your sandbox. Make sure you have the latest sandbox code by running `svn up .` in your sandbox's `/public_html` folder to get the live currency rates. 
* Delete the `wcpay_multi_currency_cached_currencies` option in order to force the client to refetch currency rates from the server.
* Set your store's default currency to `USD`. Go to `Settings -> Multi-Currency` and use the `Add Currencies` modal to add a few zero-decimal (e.g. Vietnamese Dong, Indonesian Rupiah) and a few non-zero-decimal currencies (e.g. Euros, British Pounds). Look at the enabled currencies list and verify that the rates displayed look correct, and display in the format `1.00 USD -> X.XX CAD` (for example).
* Click on the pencil icon next to a currency to go to the Edit Currency page. Verify the rate displayed is in the format ` Fetch rate automatically. Current rate: 1.00 USD = X.XX CAD (Last updated 12:00 UTC)`
* Change your store's default currency to a zero-decimal currency, e.g. Vietnamese Dong.
* Go back to the Enabled Currencies page. 
* Verify that Vietnamese Dong now displays at the top of the enabled currencies list, and `Default Currency` is displayed next to it where the conversion rate would normally display.
* Verify that the conversion rates between VND and the other enabled currencies look correct and are displayed in the format `1,000 VND -> X.XX CAD` (for example).
* Click on the pencil icon next to a currency to go to the Edit Currency page. Verify the rate displayed is in the format ` Fetch rate automatically. Current rate: 1,000 VND = X.XX CAD (Last updated 12:00 UTC)`

*Questions*

- Does the React code make sense? I am still relatively new to this so would appreciate any feedback!
- Is there a way to test this using Jest? I was considering doing something like setting up another snapshot where a zero-decimal currency was used as the default currency, and checking the format of the exchange rate displayed, but wasn't sure the best way to do this.
-------------------
